src/App.js

```diff
@@ -1,86 +1,93 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import { Header } from './components/Header/Header';
+import { TodoList } from './components/TodoList/TodoList';
+import { Footer } from './components/Footer/Footer';
+// import data from './data/data.json';
 
 function App() {
+  const [todos, setTodos] = useState([]);
+  const [isFooterVisible, setIsFooterVisible] = useState(false);
+
+  useEffect(() => {
+    const saveTodos = localStorage.getItem('todos');
+
+    if (saveTodos) {
+      setTodos(JSON.parse(saveTodos));
+    }
+  }, []);
+
+  useEffect(() => {
+    localStorage.setItem('todos', JSON.stringify(todos));
+
+    if (todos.length > 0) {
+      setIsFooterVisible(true);
+    } else {
+      setIsFooterVisible(false);
+    }
+
+    setIsFooterVisible(todos.length > 0);
+  }, [todos]);
+
+  const onAll = () => {
+    setTodos(todos);
+  };
+
+  const onActive = () => {
+    const active = [...todos].filter(t => !t.completed);
+
+    setTodos(active);
+  };
+
+  const onCompleted = () => {
+    const completed = [...todos].filter(t => t.completed);
+
+    setTodos(completed);
+  };
+
+  const onClearAll = () => {
+    const clearCompleted = [...todos].filter(t => !t.completed);
+
+    setTodos(clearCompleted);
+  };
```

Comment:
Now you are REWRITING your todos every time you're filtering them. All todos should be saved in the state, and when you're applying some filters, you should only show completed/active todos but not REWRITE them all.
If you want to see only active todos and then you want to see completed, you can't, because after applying the 'active' filter, you will write only active todos to your state and there will be no more completed